src/cloud-api-adaptor/docs/SecureComms.md

```diff
@@ -0,0 +1,98 @@
+        apt install vim
+        vim /opa/confidential-containers/kbs/policy.rego // replace default policy to true
+```
+
+
+
+### Build podvm to enforce Secure-Comms
+
+Change the `podvm/files/etc/systemd/system/agent-protocol-forwarder.service` to include:
+```sh
+ExecStart=/usr/local/bin/agent-protocol-forwarder -kata-agent-namespace /run/netns/podns -secure-comms -kata-agent-socket /run/kata-containers/agent.sock $TLS_OPTIONS $OPTIONS
+```
+
+You may also include additional Inbounds and Outbounds configurations to the Forwarder using the `-secure-comms-inbounds` and `-secure-comms-outbounds` flags. See more details regarding Inbounds and Outbounds below.
+
+### Activate Secure-Comms in peer-pods-cm
+
+Add to the `peer-pods-cm` config map at the `confidential-containers-system` namespace:
+```sh
+apiVersion: v1
+data:
+    ...
+    SECURE_COMMS: "true"
+    ...
+```
+
+You may also include additional Inbounds and Outbounds configurations to the Adaptor using the `SECURE_COMMS_INBOUNDS` and `SECURE_COMMS_OUTBOUNDS` config points. See more details regarding Inbounds and Outbounds below.
+
+You may also set the KBS address using the `SECURE_COMMS_KBS_ADDR` config point.
+
+
+### Adding named tunnels to the SSH channel
+Named tunnels can be added to the SSH channel. Adding a named tunnel requires adding an Inbound at one of the SSH channel peers and an Outbound at the other SSH channel peer. The Inbound and Outbound both carry the name of the tunnel being created.
+
+        |---------Tunnel----------| 
+Client->Inbound----------->Outbound->Server
+
+
+Inbounds and Outbounds take the form of a comma separated inbound/outbound tags such that Inbounds are formed as "InboundTag1,InboundTag2,InboundTag3,..." and Outbounds are formed as "OutboundTag1,OutboundTag2,outboundTag3,..."
+
+Each Inbound tag is structured as `Phase:Name:Port` where:
+- Phase can be 'K' to represent an inbound available during the Kubernetes phase, 'A' to represent an inbound available during the Attestation phase, or 'B' to represent an inbound available during both phases.
+- Name is the name of the tunnel
+- Port is the local service port being opened to serve as ingress of the tunnel.
+
+Each outbound tag is structured as `Phase:Name:Host:Port` or `Phase:Name:Port` where:
+- Phase can be 'K' to represent an outbound available during the Kubernetes phase, 'A' to represent an outbound available during the Attestation phase, or 'B' to represent an outbound available during both phases.
+- Name is the name of the tunnel
+- Host (if available) is a hostname or IP address of the destination host to forward the information to.
+- Port is the destination port to forward the information to.
+
+For example, an outbound tag such as `K:ABC:myhost.com:1234` means that during the `Kubernetes phase`, an output of a tunnel named `ABC` is registered, such that information from a client connecting to ABC Inbound will be tunneled and forwarded to `myhost.com` port `1234`).
```

Comment:
I think explicitly adding "tcp" label into each label will leave room for extension of future UDP support.

SSH looks like supports SOCKS tunnels which support UDP, and also supports TUN devices for VPN. I have no idea how these features are implemented in SSH, but I think UDP support seems feasible.

Reply:
UDP Support:
This is feasible and we can add that in a follow up PR. 

Tags:
We can add a T/U in front of the port number to represent TCP and UDP. 
Yet, I would prefer to leave the default to be TCP - i.e. when no T/U prefix are used, we refer to the tag as being TCP.
This leaves the common hostname:port combination as is for the common tcp case. 

In fact, we can leave tags as is for this PR and add in the next PR the options to specify the transport layer specifically such that:

`K:ABC:Myhost.com:1234`  - means tcp port 1234, in this and in future PRs
`K:ABC:Myhost.com:T1234`  - means tcp port 1234, will be added in the next PR
`K:ABC:Myhost.com:U1234`  - means udp port 1234, will be added in the next PR

Reply:
That makes sense.